SparkPost/matchbox
# UX-786 Include styled-system types as runtime dependency

### What Changed
- Moves `@types/styled-system` from a dev dependency to a runtime dependency in matchbox

### How To Test or Verify
- Verify the build works
- Very matchbox includes styled-system types when installed (tested with `^7.1.1-beta.1`)

### PR Checklist
